types/src/error.rs

```diff
@@ -145,6 +145,8 @@ pub const SERVER_IS_BUSY_CODE: i32 = -32009;
 pub const TOO_BIG_BATCH_REQUEST_CODE: i32 = -32010;
 /// Batch response limit was exceed.
 pub const TOO_BIG_BATCH_RESPONSE_CODE: i32 = -32011;
+/// Middleware error code.
+pub const MIDDLEWARE_ERROR_CODE: i32 = -32012;
```

Comment:
I'm a bit on the fence wondering whether there should be an error specific to middleware, or whether middlewares should emit "normal" errors or use custom codes or whatever. Like, when would one want to return a "middleware error" rather than an "auth error" or "rate limit error" or something?

Reply:
This shouldn't occur since we are using `Infallible` all over the place but if someone decides to return error from "their own middleware impl" this may occur.

We can probably just do a custom error and remove this, probably for the best.

Reply:
enforced this requiring Error = Infallible in the server